app/pages/sections/additional-services/order-item/flat/patient/ui-tests/general.ui.test.js

```diff
@@ -0,0 +1,258 @@
+import nock from 'nock';
+import { ClientFunction, Selector } from 'testcafe';
+import { extractInnerText } from 'buying-catalogue-library';
+import content from '../manifest.json';
+import { solutionsApiUrl } from '../../../../../../../config';
+
+const pageUrl = 'http://localhost:1234/order/organisation/order-1/additional-services/neworderitem';
+
+const getLocation = ClientFunction(() => document.location.href);
+
+const selectedPrice = {
+  priceId: 1,
+  provisioningType: 'Patient',
+  type: 'Flat',
+  currencyCode: 'GBP',
+  itemUnit: {
+    name: 'consultation',
+    description: 'per consultation',
+  },
+  price: 0.11,
+};
+
+const authTokenInSession = JSON.stringify({
+  id: '88421113', name: 'Cool Dude', ordering: 'manage', primaryOrganisationId: 'org-id',
+});
+const itemIdInSession = 'item-1';
+const selectedRecipientIdInSession = 'recipient-1';
+const selectedRecipientNameInSession = 'recipient-name';
+const selectedPriceIdInSession = 'price-1';
+
+const orderItemPageDataInSession = JSON.stringify({
+  itemId: itemIdInSession,
+  serviceRecipientId: selectedRecipientIdInSession,
+  serviceRecipientName: selectedRecipientNameInSession,
+  selectedPrice,
+});
+
+const setState = ClientFunction((key, value) => {
+  document.cookie = `${key}=${value}`;
+});
+
+const mocks = () => {
+  nock(solutionsApiUrl)
+    .get('/api/v1/catalogue-items/item-1')
+    .reply(200, { id: 'item-1', name: 'Item One' });
+  nock(solutionsApiUrl)
+    .get('/api/v1/prices/price-1')
+    .reply(200, selectedPrice);
+};
+
+const pageSetup = async (withAuth = true, postRoute = false) => {
+  if (withAuth) {
+    mocks();
+    await setState('fakeToken', authTokenInSession);
+    await setState('selectedRecipientId', selectedRecipientIdInSession);
+    await setState('selectedRecipientName', selectedRecipientNameInSession);
+    await setState('selectedItemId', itemIdInSession);
+    await setState('selectedPriceId', selectedPriceIdInSession);
+    if (postRoute) {
+      await setState('orderItemPageData', orderItemPageDataInSession);
+    }
+  }
+};
+
+fixture('Additional-services - flat patient - general')
+  .page('http://localhost:1234/order/some-fake-page')
+  .afterEach(async (t) => {
+    const isDone = nock.isDone();
+    if (!isDone) {
+      nock.cleanAll();
+    }
+
+    await t.expect(isDone).ok('Not all nock interceptors were used!');
+  });
+
```

Comment:
should there be tests to make sure the page has rendered and getting the title etc?

Reply:
yes we have them in the ui-tests at the root level to avoid doing that check 3 times for each orderItemType.

Reply:
ah right yeah!